0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -150,6 +151,11 @@ One option is to encrypt any secrets at-rest using an Authenticated Encryption a
   
 ### Secrets In Memory
 Implementations SHOULD minimize the amount of time that an actual secret-value exists in-memory in unencrypted form. This includes narrowing the availability of secrets to only code that actually requires them; minimizing the time any secret might exist in memory; and zeroing out memory after a secret is no longer used, if possible. 
+
+### Mutual TLS
+All ILP-over-HTTP connections MUST be performed over a TLS session. However, it is also RECOMMENDED to use TLS Client Certificates between peers for additional _network-level_ security. 
+
+Note that while possible, it is not advised to establish the identity of an ILP Principle using Client Certificates because many network operators will terminate TLS connections at the network edge, and then forward traffic on to actual backends in a new connection (e.g., a new TLS session or simply unencrypted). In TLS-termination scenarios, the actual backend will not be able to validate the original Client Certificate identity, so a higher-level Authentication scheme (like tokens) is required.
```

Comment:
I disagree with this as it is making big assumptions about how someone will use and deploy TLS.

There are notes here that are good for people to consider when using Client Certs but I wouldn't go so far as to say this is not advised.

On the contrary it is likely to be most efficient way of doing auth as long as the parties do it correctly. 

From what I have observed for example, many cloud providers support Client Cert **validation** and then pass the cert along as a header for the application to then perform **authentication**. 

Once this is done once by the application the result can be cached for a long time because the cert will seldom change (Certificates have much longer lifetimes than JWT's for example).

Reply:
Fair points. I've removed this section entirely. This spec doesn't really need to cover any of this other than bringing Mutual TLS to the attention of the reader (IMHO).